test/unit-test/mqtt_agent_utest.c

```diff
@@ -1416,5 +1478,39 @@ void test_MQTTAgent_CommandLoop_with_eventCallback( void )
 
     TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
     /* Ensure that callback is not invoked. */
-    TEST_ASSERT_EQUAL( 9, commandCompleteCallbackCount );
+    TEST_ASSERT_EQUAL( 10, commandCompleteCallbackCount );
+}
+
+/**
+ * @brief Test MQTTAgent_CommandLoop behavior with multiple calls to MQTT_ProcessLoop.
+ */
+void test_MQTTAgent_CommandLoop_calling_MQTTProcessLoop_multiple_times( void )
+{
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+    Command_t commandToSend;
+
+    setupAgentContext( &mqttAgentContext );
+    mqttAgentContext.mqttContext.connectStatus = MQTTConnected;
+    pReturnFlags.addAcknowledgment = false;
+    pReturnFlags.runProcessLoop = true;
+    pReturnFlags.endLoop = true;
+
+    MQTTAgentCommand_Publish_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTTAgentCommand_Publish_ReturnThruPtr_pReturnFlags( &pReturnFlags );
+    /* Calling MQTT_ProcessLoop multiple times. */
+    MQTT_ProcessLoop_StubWithCallback( MQTT_ProcessLoop_stub_multiple );
```

Comment:
`_StubWithCallback` is deprecated according to [CMock documentation](https://github.com/ThrowTheSwitch/CMock/blob/master/docs/CMock_Summary.md#callback). This should be `_Stub` if we haven't added any expect calls for `MQTT_ProcessLoop` and don't want CMock to keep track of the number of invocations; if we do, then it's `_AddCallback`
```suggestion
    MQTT_ProcessLoop_Stub( MQTT_ProcessLoop_stub_multiple );
```